scalding-core/src/main/scala/com/twitter/scalding/Config.scala

```diff
@@ -402,6 +410,14 @@ object Config {
   val PipeDescriptions = "scalding.pipe.descriptions"
   val StepDescriptions = "scalding.step.descriptions"
 
+  /**
+   * Parameter that can be used to determine behavior on the rhs of a hashJoin.
+   * If true, we skip force to disk on MapFn and FlatMapFn transforms as long as they're
+   * followed by a Pipe we think has been persisted (e.g. Checkpoint).
+   * Else we forceToDisk in those two scenarios (along with potentially others like FilterFn)
+   */
+  val SkipForceHashJoinRHS: String = "scalding.hashjoin.skipforceright"
```

Comment:
let's change this to `scalding.hashjoin.autoforceright` : if true, scalding will guess when to try to force to disk before a hashjoin, if false (the default) do exactly what the user says to do.